source/acquaman/IDEAS/IDEASScanConfiguration.h

```diff
@@ -0,0 +1,92 @@
+#ifndef IDEASSCANCONFIGURATION_H
+#define IDEASSCANCONFIGURATION_H
+
+#include "acquaman/IDEAS/IDEASScanConfigurationDbObject.h"
+#include "application/IDEAS/IDEAS.h"
+
+/*!
+	This is a non-QObject class that holds the database object for a general
+	scan configuration.  It has getters and setters to the database object.
+	*/
+class IDEASScanConfiguration
+{
+public:
+	/// Constructor.
+	IDEASScanConfiguration();
+	/// Copy constructor.
+	IDEASScanConfiguration(const IDEASScanConfiguration &original);
+	/// Destructor.
+	virtual ~IDEASScanConfiguration();
+
+	// DbObject specific methods.
+	////////////////////////////////////////////
+
+	/// Returns the database object.  Intended for gaining access to its signals.
+	IDEASScanConfigurationDbObject *dbObject() const { return dbObject_; }
+	/// The database reading member function.
+	AMDbObject *dbReadScanConfigurationDbObject() { return dbObject_; }
+	/// The database writing member function.
+	void dbWriteScanConfigurationDbObject(AMDbObject *object);
+
+	// Getters
+	/////////////////////////////////////////
+
+	/// Returns the energy.
+	double energy() const { return dbObject_->energy(); }
+	/// Returns the current fluorescence detector choice.
+	IDEAS::FluorescenceDetectors fluorescenceDetector() const { return dbObject_->fluorescenceDetector(); }
+	/// Returns the list of regions of interest.
+	QList<AMRegionOfInterest *> regionsOfInterest() const { return dbObject_->regionsOfInterest(); }
+
+	/// Returns the current total estimated time for a scan to complete.
+	double totalTime() const { return totalTime_; }
+	/// Overloaded.  Returns the current total estimated time but also specifies whether the time should be recomputed first.
+	double totalTime(bool recompute) { if (recompute) computeTotalTimeImplementation(); return totalTime_; }
```

Comment:
Likely true.  It's only like this due to historical reasons that only past Darren knows about.  They are a mystery to me now.